### DISCUSSION TITLE
吐槽Dock崩溃，所有应用强制退出。

### DISCUSSION THREAD

**Discussion by Linux-qitong** [linuxdeepin/developer-center#4334] [Troubleshooting | 故障排查] [answered]
https://bbs.deepin.org/phone/zh/post/256351
在正常编辑文档或看网页过程中，Docker经常出现崩溃（消失）几秒钟后，Docker重新出现，所有之前运行的程序（微信、QQ、浏览器、文档）全部强制退出，之前做的东西全部丢失。超级郁闷……

**Comment by BLumia** [comment]
dock 崩溃不会导致其它应用全部强制退出呀，你开个终端 `killall dde-dock` 试试就知道了，无论 v20 还是 23，dock 崩溃都不会退掉别的应用。

需要收集下更多信息看看，如何让别人复现这个问题。

**Comment by BLumia** [comment]
可以 `coredumpctl list` 看下问题发生后，最新崩溃的应用里是不是有 dde-application-manager ，有的话考虑提供下 dde-application-manager 对应的 coredump stacktrace。（需要先安装这个包的调试包，并同时提供 dde-application-manager 的版本信息）

**Comment by BLumia** [comment, marked as answer]
根据 bbs 帖子的后续信息，推测应该是 application-manager 崩溃导致的问题，应该会在 beta 前修复（已有相关修复合入）。

参见：

- https://github.com/linuxdeepin/dde-application-manager/pull/45
- https://github.com/linuxdeepin/developer-center/issues/3598

**Reply by ice909** [reply]
好像并没有完全修复，我这里还是频繁崩溃

![image](https://github.com/linuxdeepin/developer-center/assets/59855569/7e804698-5af8-4688-ac52-7c53d038517d)

**Reply by Decodetalkers** [reply]
> 好像并没有完全修复，我这里还是频繁崩溃 ![image](https://user-images.githubusercontent.com/59855569/240500402-7e804698-5af8-4688-ac52-7c53d038517d.png)

这个项目比较麻烦，能查下coredump的位置吗。现在能做的是尽可能减少coredump

**Reply by ice909** [reply]
是这个吗？

![image](https://github.com/linuxdeepin/developer-center/assets/59855569/22c659fe-4952-41df-9f12-72a1e42b69d3)

**Reply by BLumia** [reply]
是。顺便，有比较稳定的复现步骤吗？

另外，通过全局搜索启动的应用应该不会因 AM 崩溃而被退出。如果希望临时绕过问题的话可以这么启应用。

**Reply by ice909** [reply]
我这里有一个稳定复现的方法：

首先在`控制中心-个性化-桌面` 勾选`显示最近使用的应用`

然后使用vscode打开下面这个demo
[demo2.zip](https://github.com/linuxdeepin/developer-center/files/11562170/demo2.zip)

重复进行`运行demo`，`关闭运行起来的demo窗口` 操作，AM就会崩溃。

**Reply by BLumia** [reply]
cc @black-desk

**Reply by black-desk** [reply]
麻烦把整个coredump发过来 这样光看堆栈会丢失不少信息。

**Reply by ice909** [reply]
[core.dde-application.1000.41588d606af34dbeae066ec40684375a.40218.1684909587000000.zip](https://github.com/linuxdeepin/developer-center/files/11563267/core.dde-application.1000.41588d606af34dbeae066ec40684375a.40218.1684909587000000.zip)
